File: php/src/Structural/Bridge/SodaImpSingleton.php
<?

#//SodaImpSingleton.java - a Singleton to hold the current SodaImp

class SodaImpSingleton 
{

	static $sodaImp;

	function setTheSodaImp($si)
	{
		self::$sodaImp = $si;
	}
    
	function getTheSodaImp ()
	{
		return self::$sodaImp;
	}

}

File: python/src/Structural/Bridge/SodaImpSingleton.py

#//SodaImpSingleton.java - a Singleton to hold the current SodaImp

class SodaImpSingleton:

	_sodaimp = None
	
	def __init__(self,sodaimp):
		SodaImpSingleton._sodaimp = sodaimp

	@staticmethod
	def getTheSodaImp() : 
		return SodaImpSingleton._sodaimp

